diff --git a/utils/bazel/llvm-project-overlay/compiler-rt/BUILD.bazel b/utils/bazel/llvm-project-overlay/compiler-rt/BUILD.bazel
index bcf947a23014..7e9bfb8a19ca 100644
--- a/utils/bazel/llvm-project-overlay/compiler-rt/BUILD.bazel
+++ b/utils/bazel/llvm-project-overlay/compiler-rt/BUILD.bazel
@@ -111,3 +111,15 @@ cc_library(
         ":orc_rt_common_headers",
     ],
 )
+
+filegroup(
+    name = "msan_ignorelist",
+    srcs = ["lib/msan/msan_ignorelist.txt"],
+    visibility = ["//visibility:public"],
+)
+
+filegroup(
+    name = "asan_ignorelist",
+    srcs = ["lib/asan/asan_ignorelist.txt"],
+    visibility = ["//visibility:public"],
+)
