--- a/utils/bazel/llvm-project-overlay/llvm/driver.bzl
+++ b/utils/bazel/llvm-project-overlay/llvm/driver.bzl
@@ -128,9 +128,12 @@
     # This is consistent with how tools/llvm-driver/CMakeLists.txt does it,
     # and this makes sure that more specific tools are checked first.
     # For example, "clang-scan-deps" should not match "clang".
-    tools = [label_to_name[tool.label.name] for tool in ctx.attr.driver_tools]
+    tools = sorted(
+        [label_to_name[tool.label.name] for tool in ctx.attr.driver_tools],
+        reverse = True,
+    )
     tool_alias_pairs = []
-    for tool_name in reversed(tools):
+    for tool_name in tools:
         tool_alias_pairs.append((tool_name, tool_name))
         for extra_alias in _EXTRA_ALIASES.get(tool_name, []):
             tool_alias_pairs.append((tool_name, extra_alias))
