diff --git a/mingw-w64-crt/lib-common/msvcrt.def.in b/mingw-w64-crt/lib-common/msvcrt.def.in
index 95b25012b..95b25012b 100644
--- a/mingw-w64-crt/lib-common/msvcrt.def.in
+++ b/mingw-w64-crt/lib-common/msvcrt.def.in
@@ -603,7 +603,6 @@
 _fmode DATA
 _fpclass
 F_NON_X64(_fpieee_flt)
-_fpreset F_LD80(DATA)
 _fputchar
 _fputwchar
 _fsopen
diff --git a/mingw-w64-crt/lib-common/api-ms-win-crt-runtime-l1-1-0.def.in b/mingw-w64-crt/lib-common/api-ms-win-crt-runtime-l1-1-0.def.in
index 1c46c4f0b..1c46c4f0b 100644
--- a/mingw-w64-crt/lib-common/api-ms-win-crt-runtime-l1-1-0.def.in
+++ b/mingw-w64-crt/lib-common/api-ms-win-crt-runtime-l1-1-0.def.in
@@ -44,7 +44,6 @@
 _execute_onexit_table
 _exit
 _fpieee_flt
-_fpreset F_LD80(DATA) ; DATA added manually
 _get_doserrno
 _get_errno
 _get_initial_narrow_environment
