diff --git a/rs/experimental/toolchains/declare_rustc_toolchains.bzl b/rs/experimental/toolchains/declare_rustc_toolchains.bzl
--- a/rs/experimental/toolchains/declare_rustc_toolchains.bzl
+++ b/rs/experimental/toolchains/declare_rustc_toolchains.bzl
@@ -76,7 +76,9 @@ def declare_rustc_toolchains(
             stdlib_linkflags = select({
                 "@platforms//os:freebsd": ["-lexecinfo", "-lpthread"],
                 "@platforms//os:macos": ["-lSystem", "-lresolv"],
                 # TODO: windows
                 "//conditions:default": [],
             }),
+            extra_exec_rustc_flags = ["-Cstrip=none"],
+            extra_rustc_flags = ["-Cstrip=none"],
             default_edition = edition,
             exec_triple = triple,
