diff --git a/cc/private/link/lto_backends.bzl b/cc/private/link/lto_backends.bzl
index 5882903..2cef9da 100644
--- a/cc/private/link/lto_backends.bzl
+++ b/cc/private/link/lto_backends.bzl
@@ -475,6 +475,7 @@ def _create_lto_backend_action(
     outputs = _get_lto_backend_action_outputs(object_file, dwo_file)
 
     _path_variables = _paths_build_variables(
+        feature_configuration,
         index,
         object_file,
         dwo_file,
@@ -495,7 +496,7 @@ def _create_lto_backend_action(
         env = env,
     )
 
-def _paths_build_variables(index, object_file, dwo_file, bitcode_file):
+def _paths_build_variables(feature_configuration, index, object_file, dwo_file, bitcode_file):
     build_variables = {}
 
     # Ideally, those strings would come directly from the execPath of the Artifacts of
@@ -507,7 +508,9 @@ def _paths_build_variables(index, object_file, dwo_file, bitcode_file):
         build_variables["thinlto_index"] = index
     else:
         # An empty input indicates not to perform cross-module optimization.
-        build_variables["thinlto_index"] = "/dev/null"
+        build_variables["thinlto_index"] = (
+            "NUL" if feature_configuration.is_enabled("targets_windows") else "/dev/null"
+        )
 
     # The output from the LTO backend step is a native object file.
     build_variables["thinlto_output_object_file"] = object_file
diff --git a/cc/toolchains/args/libraries_to_link/BUILD b/cc/toolchains/args/libraries_to_link/BUILD
index 60e59f7..ebfec9a 100644
--- a/cc/toolchains/args/libraries_to_link/BUILD
+++ b/cc/toolchains/args/libraries_to_link/BUILD
@@ -56,7 +56,10 @@ cc_nested_args(
 
 cc_nested_args(
     name = "start_lib_arg",
-    args = ["-Wl,--start-lib"],
+    args = select({
+        "@platforms//os:windows": ["-Wl,/start-lib"],
+        "//conditions:default": ["-Wl,--start-lib"],
+    }),
     requires_false = "//cc/toolchains/variables:libraries_to_link.is_whole_archive",
 )
 
@@ -69,7 +72,10 @@ cc_nested_args(
 
 cc_nested_args(
     name = "end_lib_arg",
-    args = ["-Wl,--end-lib"],
+    args = select({
+        "@platforms//os:windows": ["-Wl,/end-lib"],
+        "//conditions:default": ["-Wl,--end-lib"],
+    }),
     requires_false = "//cc/toolchains/variables:libraries_to_link.is_whole_archive",
 )
 
diff --git a/cc/toolchains/args/thin_lto/BUILD b/cc/toolchains/args/thin_lto/BUILD
index be6ea28..40bbc9f 100644
--- a/cc/toolchains/args/thin_lto/BUILD
+++ b/cc/toolchains/args/thin_lto/BUILD
@@ -44,9 +44,10 @@ cc_args(
     actions = [
         ":lto_index_actions",
     ],
-    args = [
-        "-Wl,-plugin-opt,thinlto-index-only={thinlto_indexing_param_file}",
-    ],
+    args = select({
+        "@platforms//os:windows": ["-Wl,--thinlto-index-only={thinlto_indexing_param_file}"],
+        "//conditions:default": ["-Wl,-plugin-opt,thinlto-index-only={thinlto_indexing_param_file}"],
+    }),
     format = {
         "thinlto_indexing_param_file": "//cc/toolchains/variables:thinlto_indexing_param_file",
     },
@@ -58,9 +59,10 @@ cc_args(
     actions = [
         ":lto_index_actions",
     ],
-    args = [
-        "-Wl,-plugin-opt,thinlto-emit-imports-files",
-    ],
+    args = select({
+        "@platforms//os:windows": ["-Wl,--thinlto-emit-imports-files"],
+        "//conditions:default": ["-Wl,-plugin-opt,thinlto-emit-imports-files"],
+    }),
 )
 
 cc_args(
@@ -68,9 +70,10 @@ cc_args(
     actions = [
         ":lto_index_actions",
     ],
-    args = [
-        "-Wl,-plugin-opt,thinlto-prefix-replace={thinlto_prefix_replace}",
-    ],
+    args = select({
+        "@platforms//os:windows": ["-Wl,--thinlto-prefix-replace={thinlto_prefix_replace}"],
+        "//conditions:default": ["-Wl,-plugin-opt,thinlto-prefix-replace={thinlto_prefix_replace}"],
+    }),
     format = {
         "thinlto_prefix_replace": "//cc/toolchains/variables:thinlto_prefix_replace",
     },
@@ -82,9 +85,10 @@ cc_args(
     actions = [
         ":lto_index_actions",
     ],
-    args = [
-        "-Wl,-plugin-opt,thinlto-object-suffix-replace={thinlto_object_suffix_replace}",
-    ],
+    args = select({
+        "@platforms//os:windows": ["-Wl,--thinlto-object-suffix-replace={thinlto_object_suffix_replace}"],
+        "//conditions:default": ["-Wl,-plugin-opt,thinlto-object-suffix-replace={thinlto_object_suffix_replace}"],
+    }),
     format = {
         "thinlto_object_suffix_replace": "//cc/toolchains/variables:thinlto_object_suffix_replace",
     },
@@ -96,9 +100,10 @@ cc_args(
     actions = [
         ":lto_index_actions",
     ],
-    args = [
-        "-Wl,-plugin-opt,obj-path={thinlto_merged_object_file}",
-    ],
+    args = select({
+        "@platforms//os:windows": ["-Wl,/lto-obj-path:{thinlto_merged_object_file}"],
+        "//conditions:default": ["-Wl,-plugin-opt,obj-path={thinlto_merged_object_file}"],
+    }),
     format = {
         "thinlto_merged_object_file": "//cc/toolchains/variables:thinlto_merged_object_file",
     },
