diff --git a/rust/platform/triple_mappings.bzl b/rust/platform/triple_mappings.bzl
index 05ca7ea3..0ddf07f2 100644
--- a/rust/platform/triple_mappings.bzl
+++ b/rust/platform/triple_mappings.bzl
@@ -438,2 +438,4 @@
-def system_to_staticlib_ext(system):
-    return _SYSTEM_TO_STATICLIB_EXT[system]
+def system_to_staticlib_ext(system, abi = None):
+    if system == "windows" and abi in ("gnu", "gnullvm"):
+        return ".a"
+    return _SYSTEM_TO_STATICLIB_EXT[system]
diff --git a/rust/private/repository_utils.bzl b/rust/private/repository_utils.bzl
index 40775d8c..53609865 100644
--- a/rust/private/repository_utils.bzl
+++ b/rust/private/repository_utils.bzl
@@ -117 +117 @@
-        staticlib_ext = system_to_staticlib_ext(target_triple.system),
+        staticlib_ext = system_to_staticlib_ext(target_triple.system, target_triple.abi),
@@ -126 +126 @@
-                staticlib_ext = system_to_staticlib_ext(target_triple.system),
+                staticlib_ext = system_to_staticlib_ext(target_triple.system, target_triple.abi),
@@ -333 +333 @@
-        staticlib_ext = system_to_staticlib_ext(target_triple.system),
+        staticlib_ext = system_to_staticlib_ext(target_triple.system, target_triple.abi),
@@ -465 +465 @@
-        staticlib_ext = system_to_staticlib_ext(target_triple.system),
+        staticlib_ext = system_to_staticlib_ext(target_triple.system, target_triple.abi),
diff --git a/rust/private/utils.bzl b/rust/private/utils.bzl
index 3e0a7d65..cbca38f4 100644
--- a/rust/private/utils.bzl
+++ b/rust/private/utils.bzl
@@ -866 +866 @@
-    if toolchain.target_triple and toolchain.target_os == "windows" and crate_type not in ("lib", "rlib"):
+    if toolchain.target_triple and toolchain.target_os == "windows" and toolchain.target_abi not in ("gnu", "gnullvm") and crate_type not in ("lib", "rlib"):
