diff --git a/cc/toolchains/args/strip_flags/BUILD b/cc/toolchains/args/strip_flags/BUILD
index 8b61d77..a3d9b25 100644
--- a/cc/toolchains/args/strip_flags/BUILD
+++ b/cc/toolchains/args/strip_flags/BUILD
@@ -43,6 +43,7 @@ cc_args(
     name = "preserve_dates",
     actions = ["//cc/toolchains/actions:strip"],
     args = select({
         "@platforms//os:macos": [],
+        "@platforms//os:windows": [],
         "//conditions:default": ["-p"],
     }),
 )
